.github/workflows/labeling.yml

```diff
@@ -0,0 +1,24 @@
+# This workflow adds the 'ci-passed' label to a pull request once the 'CI Check' workflow completes successfully.
+
+name: Labeling Workflow
+
+on:
+  workflow_run:
+    workflows: ["CI Check"]
+    types:
+      - completed
+
+permissions:
+  pull-requests: write
+  checks: read
+
+jobs:
+  add_ci_passed_label:
+    if: github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-latest    
+    steps:
+      - name: Add 'ci-passed' label
+        run: |
+          gh pr edit ${{ github.event.workflow_run.pull_requests[0].number }} --add-label "ci-passed" --repo $GITHUB_REPOSITORY
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
How about doing this in `.github/workflows/ci-checks.yml`?
You can move the `add_ci_passed_label` and declare that it depends on `check_ci_status`:

```yaml
  add_ci_passed_label:
    needs: [check_ci_status]
```
Then you can remove `if: github.event.workflow_run.conclusion == 'success'`.